crates/oxc_linter/src/lib.rs

```diff
@@ -9,22 +9,47 @@ mod context;
 mod disable_directives;
 mod fixer;
 mod globals;
+mod lint_adapter;
 pub mod rule;
 mod rules;
 
-use std::{fs, io::Write, rc::Rc};
+use std::{collections::BTreeMap, fs, io::Write, rc::Rc, sync::Arc};
 
-pub use fixer::{FixResult, Fixer, Message};
+pub use fixer::{Fixer, Message};
+use lazy_static::lazy_static;
+use lint_adapter::{InputQuery, LintAdapter};
+use miette::{miette, LabeledSpan};
 pub(crate) use oxc_semantic::AstNode;
 use oxc_semantic::Semantic;
 use rustc_hash::FxHashMap;
+use trustfall::{execute_query, Schema, TransparentValue};
 
 use crate::context::LintContext;
 pub use crate::{
     rule::RuleCategory,
     rules::{RuleEnum, RULES},
 };
 
+lazy_static! {
+    static ref TRUSTFALL_RULES: Vec<InputQuery> = fs::read_dir("queries")
+        .unwrap()
+        .filter_map(std::result::Result::ok)
+        .filter(|dir_entry| dir_entry.path().is_file())
+        .filter(|f| f.path().as_os_str().to_str().unwrap().ends_with(".ron"))
+        .map(|f| fs::read_to_string(f.path()))
+        .map(std::result::Result::unwrap)
+        .map(|rule| { ron::from_str::<InputQuery>(rule.as_str()).unwrap() })
+        .collect();
+    static ref SCHEMA: Schema = match Schema::parse(fs::read_to_string("./schema.graphql").unwrap())
+    {
+        Ok(schema) => schema,
+        Err(err) => {
+            println!("{}", err);
+            unimplemented!()
+        }
+    };
+}
```

Comment:
Just use `expect()` here, it does the same thing as the match but in fewer lines.
```suggestion
    static ref SCHEMA: Schema = Schema::parse(fs::read_to_string("./schema.graphql").expect("failed to read schema file"))
        .expect("schema file failed to parse");
}
```

Reply:
Done, thanks!